### ISSUE TITLE
CLIP temperature scaling

### ISSUE THREAD

**Issue by Do-minik** [BAAI-DCAI/M3D#52]
Hello, thank you for your work and publishing the code.

I have a question regarding the temperature scaling in your CLIP.py script.
You are initializing the temperature in the log scale:

```
self.logit_scale = nn.Parameter(torch.ones([]) * np.log(1 / 0.07))
```
Then, you are directly scaling the similarity values by this log-scale temperature.
```
logits_per_image = self.logit_scale * image_features @ text_features.T
```

Wouldn't it be necessary to exponentiate the logit_scale to conform to the CLIP loss? In my opinion,
the line should be:
```
logits_per_image = self.logit_scale.exp() * image_features @ text_features.T
```
This way, we should scale the similarity values by 1/temperature like in the original CLIP loss and not by log(1/temperature).

I verified in the original CLIP repository: 
init: https://github.com/openai/CLIP/blob/main/clip/model.py#L295
scale: https://github.com/openai/CLIP/blob/main/clip/model.py#L367

I would be happy if you could point out the error in my thought process or confirm if it is correct.

**Comment by baifanxxx** [comment]
Hi,

Thank you for your interest in our work. You are very observant—indeed, checking the source code confirms that self.logit_scale.exp() is correct. However, since self.logit_scale is a learnable parameter, whether or not to apply .exp() is technically valid and makes little practical difference. Additionally, its initialization value has minimal impact on the outcome.

Best regards,
BAI Fan
